Source/WTF/wtf/HashSet.h

```diff
@@ -132,6 +132,33 @@ class HashSet final {
     TakeType take(iterator);
     TakeType takeAny();
 
+    // Returns a new set with the elements of both this and the given set.
+    template<typename OtherValueArg, typename OtherHashArg, typename OtherTraitsArg, typename OtherTableTraitsArg>
+    HashSet unionWith(const HashSet<OtherValueArg, OtherHashArg, OtherTraitsArg, OtherTableTraitsArg>&) const;
+
+    // Returns a new set with the elements that are common to both this set and
+    // the given set.
+    template<typename OtherValueArg, typename OtherHashArg, typename OtherTraitsArg, typename OtherTableTraitsArg>
+    HashSet intersectionWith(const HashSet<OtherValueArg, OtherHashArg, OtherTraitsArg, OtherTableTraitsArg>&) const;
+
+    // Returns a new set with the elements that are either in this set or in the
+    // given set, but not in both
+    template<typename OtherValueArg, typename OtherHashArg, typename OtherTraitsArg, typename OtherTableTraitsArg>
+    HashSet symmetricDifferenceWith(const HashSet<OtherValueArg, OtherHashArg, OtherTraitsArg, OtherTableTraitsArg>&) const;
+
+    // Adds the elements of the given set to the set.
+    template<typename OtherValueArg, typename OtherHashArg, typename OtherTraitsArg, typename OtherTableTraitsArg>
+    void formUnion(const HashSet<OtherValueArg, OtherHashArg, OtherTraitsArg, OtherTableTraitsArg>&);
+
+    // Removes the elements of this set that aren't also in the given set.
+    template<typename OtherValueArg, typename OtherHashArg, typename OtherTraitsArg, typename OtherTableTraitsArg>
+    void formIntersection(const HashSet<OtherValueArg, OtherHashArg, OtherTraitsArg, OtherTableTraitsArg>&);
```

Comment:
I wish the argument type could be a bit more general. This works with any collection that has a `contains` operation, like a `ListHashSet` would be fine or a view on the keys of a `HashMap`.